opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectCreation.js

```diff
@@ -269,379 +115,278 @@ const sharedUpdater = (
   ConnectionHandler.insertEdgeBefore(conn, newEdge);
 };
 
-class StixDomainObjectCreation extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { open: false };
-  }
+const buildEntityTypes = (t, queryData, stixDomainObjectTypes) => {
+  const choices = (queryData.sdoTypes?.edges ?? []).map((edge) => ({
+    label: t(`entity_${edge.node.label}`),
+    value: edge.node.label,
+    type: edge.node.label,
+  }));
+  const entitiesTypes = R.sortWith([R.ascend(R.prop('label'))], choices);
+  return entitiesTypes.filter((n) => {
+    if (!stixDomainObjectTypes || stixDomainObjectTypes.length === 0 || stixDomainObjectTypes.includes('Stix-Domain-Object')) {
+      return !UNSUPPORTED_TYPES.includes(n.value);
+    }
+    if (stixDomainObjectTypes.includes('Identity') && IDENTITY_ENTITIES.includes(n.value)) {
+      return true;
+    }
+    if (stixDomainObjectTypes.includes('Location') && LOCATION_ENTITIES.includes(n.value)) {
+      return true;
+    }
+    return !!stixDomainObjectTypes.includes(n.value);
+  });
+};
 
-  handleOpen() {
-    this.setState({ open: true });
-  }
+const StixDomainPanel = ({ queryRef, stixDomainObjectTypes, onClose, creationUpdater,
+  confidence, defaultCreatedBy, defaultMarkingDefinitions }) => {
+  const { t } = useFormatter();
+  const queryData = usePreloadedQuery(stixDomainObjectCreationAllTypesQuery, queryRef);
+  const availableEntityTypes = buildEntityTypes(t, queryData, stixDomainObjectTypes);
+  const [status, setStatus] = useState(availableEntityTypes.at(0).value);
```

Comment:
Why this property is status ? It could be clearer to name it entity no ?

Reply:
Fixed